ogr/ogrsf_frmts/gtfs/ogrgtfsdriver.cpp

```diff
@@ -0,0 +1,724 @@
+/***********************************************************************/
+
+void OGRGTFSLayer::ResetReading()
+{
+    m_poUnderlyingLayer->ResetReading();
+}
+
+/***********************************************************************/
+/*                        PrepareTripsData()                           */
+/***********************************************************************/
+
+void OGRGTFSLayer::PrepareTripsData()
+{
+    m_bPrepared = true;
+    try
+    {
+        {
+            auto poStopsDS = std::unique_ptr<GDALDataset>(GDALDataset::Open(
+                (m_osDirname + "/stops.txt").c_str(), GDAL_OF_VECTOR));
+            if (!poStopsDS)
+                return;
+            auto poStopsLyr = poStopsDS->GetLayer(0);
+            if (!poStopsLyr)
+                return;
+            const auto poStopsLyrDefn = poStopsLyr->GetLayerDefn();
+            const int nStopIdIdx = poStopsLyrDefn->GetFieldIndex("stop_id");
+            const int nStopLatIdx = poStopsLyrDefn->GetFieldIndex("stop_lat");
+            const int nStopLonIdx = poStopsLyrDefn->GetFieldIndex("stop_lon");
+            if (nStopIdIdx < 0 || nStopLatIdx < 0 || nStopLonIdx < 0)
+                return;
+            for (auto &&poFeature : poStopsLyr)
+            {
+                const char *pszStopId = poFeature->GetFieldAsString(nStopIdIdx);
+                if (pszStopId)
+                {
+                    m_oMapStopIdToLonLat[pszStopId] = std::make_pair(
+                        poFeature->GetFieldAsDouble(nStopLonIdx),
+                        poFeature->GetFieldAsDouble(nStopLatIdx));
+                }
+            }
+        }
+
+        auto poStopTimesDS = std::unique_ptr<GDALDataset>(GDALDataset::Open(
+            (m_osDirname + "/stop_times.txt").c_str(), GDAL_OF_VECTOR));
+        if (!poStopTimesDS)
+            return;
+        auto poStopTimesLyr = poStopTimesDS->GetLayer(0);
+        if (!poStopTimesLyr)
+            return;
+        const auto poStopTimesLyrDefn = poStopTimesLyr->GetLayerDefn();
+        const int nStopIdIdx = poStopTimesLyrDefn->GetFieldIndex("stop_id");
+        const int nTripIdIdx = poStopTimesLyrDefn->GetFieldIndex("trip_id");
+        const int nStopSequenceIdx =
+            poStopTimesLyrDefn->GetFieldIndex("stop_sequence");
+        if (nStopIdIdx < 0 || nTripIdIdx < 0 || nStopSequenceIdx < 0)
+            return;
+        for (auto &&poFeature : poStopTimesLyr)
+        {
+            const char *pszStopId = poFeature->GetFieldAsString(nStopIdIdx);
+            const char *pszTripId = poFeature->GetFieldAsString(nTripIdIdx);
+            const int nStopSequence =
+                poFeature->GetFieldAsInteger(nStopSequenceIdx);
+            if (pszStopId && pszTripId)
+            {
+                m_oMapTripIdToStopIds[pszTripId][nStopSequence] = pszStopId;
+            }
+        }
+    }
+    catch (const std::exception &)
+    {
+        CPLError(CE_Failure, CPLE_OutOfMemory, "Not enough memory");
```

Comment:
Just wondering: is OOM the only case when the `try` above can `throw` ?

Reply:
I think so. The risk is here is too many objects beeing put in the std::map. I'll change it to specifically catch std::bad_alloc though